Create Virtual Network PS Script using template.json

sunshinejen opened the pull request:
Task #71 
This script creates a Virtual Network with one subnet

1. Open Powershell
2. Make sure you navigate to where script is located in the automation/vnet directory and enter ./deploy-vnet.ps1
-Enter information that is prompted, you will need an existing Azure subscription and resource group name
**You will be asked to supply values for the following parameters:**
- TentantID
- Username and Password
If you enter a resource group that does not exist 
**You will be asked to supply values for the following parameters:**
-Location: Location where new Resource group
will be deployed  

If you enter a resource group that exists
**You will be asked to supply values for the following parameters:**

- ResourceGroupName: Name of your existing resource group
- VirtualNetworkName: Name of your vnet
- SubNetworkName: Name of your subnet

On Azure confirm Virtual Network has been created with your given parameters
If you are testing make sure to delete any created resources

jennvillacis committed:
updating deploy.vnet file and adding template file

adding template.json

jennvillacis committed:
updating script and template

working with hardcoded values in variables for TemplateParameterObject, adds to azure resource

jennvillacis committed:
update deploy_vnet.ps1

script now uses existing resource location for variable location

jennvillacis committed:
updates to deploy-vnet.ps1

adds dynamic name values

jennvillacis committed:
script updates

variable name changes

jennvillacis committed:
updated template and ps1 script

script done

jennvillacis committed:
updates to script

updated to correct azure login using Service principal, and tenantid

jennvillacis committed:
updates

fixed parameter description

OyunominJuno commented on automation/vnet/deploy-vnet.ps1:
```diff
@@ -4,55 +4,70 @@
-    Required. Name of desired Virtual Network Name Eg: [project]-[resource-type]-[environment]-[location]-[other-stuff]
-.PARAMETER VNetPrefix
-    Required. Desired address space for Virtual Network eg: eg: 10.0.0.0/16
+.PARAMETER TenantId
```
in the documentation of the PR, this parameter was not mentioned.

sunshinejen replied:
Good catch, it is updated thanks Juno

OyunominJuno commented on automation/vnet/deploy-vnet.ps1:
```diff
@@ -4,55 +4,70 @@
 [cmdletbinding()]
 Param(
-    [Parameter(Mandatory=$true)]
+    [Parameter(Mandatory=$true,HelpMessage='Enter your TenantId')]
```
it is good to know about the HelpMessage. will use that in the future

OyunominJuno commented on automation/vnet/deploy-vnet.ps1:
```diff
@@ -4,55 +4,70 @@
+#if errorVariable is notPresent VNet will not be created
+if ($notPresent)
+{
+    Write-Output "Existing resource group not found"
```
if the resource group is not found, you can create one with the name entered and ask for a location. You can refer to my code for that.

OyunominJuno commented on automation/vnet/deploy-vnet.ps1:
```diff
@@ -4,55 +4,70 @@
+$RGLocation= $Location.location
+
+#file path for -TemplateFile
+$FilePath = "./template.json"
```
what if the user wants to use different template.json? you can consider asking the path from the user

OyunominJuno commented on automation/vnet/deploy-vnet.ps1:
```diff
@@ -4,55 +4,70 @@
+
+#Hashtable containing parameters for virtual network template
+$VirtualNetworkParameters= @{
+    vnetName = ("nsc-vnet-dev-{0}-{1}-test" -f $RGLocation,$VirtualNetworkName).ToLower();
```
hard coded name here, is there a way to ask this from the user?

jennvillacis committed:
update deploy script

added validation parameters, along with ability to create a new resource group if resource group is not found

elb-dev commented:
Code looks nice and readable, and the script works! It's a lot easier to follow what to do this time around, as opposed to last time. I have just one concern: I think it would make more sense for something like the two names to be set to not required, so they're easy to deploy with default parameters as required.

elb-dev reviewed (approved):
Ohp, already left a comment. Check what I said previously. I think it is good though.

asiagromasia commented on automation/vnet/template.json:
```diff
@@ -0,0 +1,68 @@
+      }
+    },
+    "location": {
+      "type": "string",
```
adding here allowed values option would limit user input and make it more safe

sunshinejen replied:
Great idea! I have added your requested changes.

OyunominJuno reviewed (approved):
this script successfully creates a virtual network and subnet within a resource group.

jennvillacis committed:
updates to template.json

adding allowedValues as suggested

asiagromasia reviewed (approved):
easy to read and deploys vnet with subnet
